src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -103,7 +103,11 @@ public FileStorage(final Path path, final FileSystem fls) {
                         .filter(Files::isRegularFile)
                         .map(Path::toString)
                         .map(p -> p.substring(dirnamelen))
-                        .map(s -> s.split(FileSystems.getDefault().getSeparator()))
+                        .map(
+                            s -> s.split(
+                                FileSystems.getDefault().getSeparator().replace("\\", "\\\\")
```

Comment:
@olegmoz could you give me the case, when replacement is needed. Cannot come up with it.

Reply:
@olegmoz sure! Thing is `String.split()` method accepts regex as parameter, not just string. When file system separator is `\` then it's illegal regex and has to be converted into '\\'. That is why I needed to add the replacement.

Reply:
@olegmoz now it's clear for me, thank you so much!